### PR TITLE
For Perlmutter, with gnu v10, add compiler flag to work-around issue with MMF tests

### PR DESCRIPTION
**Pull request by ndkeen**
Add a flag to disable new behavior in gnu v10. For now, just do this only for Perlmutter and gnu v10+.

This PR adds to new files to be specific to Perlmutter for now.

Flag itself does not change results, but the tests were hitting errors without it.

Fixes https://github.com/E3SM-Project/E3SM/issues/4824

### CODE REVIEW

**Comment by ndkeen** [comment]
merged to next

File: cime_config/machines/Depends.perlmutter.gnu.cmake [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+if (CMAKE_Fortran_COMPILER_VERSION VERSION_GREATER_EQUAL 10)
+
+  if (NOT DEBUG)
+    # new in gnu10, inline arg packing was causing INF values with SMS_P4x1.ne4pg2_ne4pg2.F-MMFXX
+    foreach(ITEM IN LISTS NO_INLINE_ARG_PACKING_LIST)
+      e3sm_add_flags("${ITEM}" " -fno-inline-arg-packing")
+    endforeach()
+  endif()
+
+endif()
```
Comment (amametjanov) [inline cime_config/machines/Depends.perlmutter.gnu.cmake:14]: I think it'd be better to move this to `Depends.gnu[gpu].cmake` and remove several duplicated mach-specific files.

Comment (ndkeen) [reply cime_config/machines/Depends.perlmutter.gnu.cmake:14]: At the time, I wasn't sure if this was something specific to the machine or compiler install. Are you seeing this helps on other machines?

Comment (amametjanov) [reply cime_config/machines/Depends.perlmutter.gnu.cmake:14]: Yes, on Crusher with gcc/11.2.0: doing this in commit https://github.com/E3SM-Project/E3SM/pull/5132/commits/b25627187cc3bc6718c3ec99449af4cb071959d4 as part of #5132

Comment (ndkeen) [reply cime_config/machines/Depends.perlmutter.gnu.cmake:14]: ok i made an issue https://github.com/E3SM-Project/E3SM/issues/5136

**Pull request merged**